packages/dendron-cli/src/commands/podsV2.ts

```diff
@@ -0,0 +1,279 @@
+        engine,
+        vaultName: args.vault,
+        fname: args.fname,
+      });
+      break;
+    case PodExportScope.Hierarchy:
+      payload = getHierarchyProps({
+        engine,
+        hierarchy: args.hierarchy,
+        vaultName: args.vault,
+      });
+      break;
+    default:
+      throw new DendronError({
+        message: `the CLI doesn't support the provided export scope: ${configValues.exportScope}. please run this export pod using the Dendon plugin`,
+      });
+  }
+  return {
+    data: {
+      ...args,
+      ...engineArgs,
+      config: configValues,
+      payload,
+    },
+  };
+}
+
+const getWorkspaceProps = (engine: DEngineClient): NoteProps[] => {
+  return Object.values(engine.notes).filter((notes) => notes.stub !== true);
+};
+
+const getVaultProps = (opts: {
+  engine: DEngineClient;
+  vaultName?: string;
+}): NoteProps[] => {
+  const { engine, vaultName } = opts;
+  const vault = checkVaultArgs({ engine, vaultName });
+  return Object.values(engine.notes).filter(
+    (note) => note.stub !== true && VaultUtils.isEqualV2(note.vault, vault)
+  );
+};
+
+const getNoteProps = (opts: {
+  engine: DEngineClient;
+  vaultName?: string;
+  fname?: string;
+}): NoteProps[] => {
+  const { engine, fname, vaultName } = opts;
+  const vault = checkVaultArgs({ engine, vaultName });
+
+  if (!fname) {
+    throw new DendronError({
+      message: "Please provide fname of note in --fname arg",
+    });
+  }
+  const note = NoteUtils.getNoteByFnameFromEngine({ fname, vault, engine });
+  if (!note)
+    throw new DendronError({
+      message: `Cannot find note with fname ${fname} in vault ${vault}`,
+    });
+  return [note];
+};
+
+// returns notes within a hierarchy (for a specefic vault)
+const getHierarchyProps = (opts: {
+  engine: DEngineClient;
+  hierarchy?: string;
+  vaultName?: string;
+}): NoteProps[] => {
+  const { engine, hierarchy, vaultName } = opts;
+  if (!hierarchy) {
+    throw new DendronError({
+      message: "Please provide hierarchy in --hierarchy arg",
+    });
+  }
+  const vault = checkVaultArgs({ engine, vaultName });
+  return Object.values(engine.notes).filter(
+    (value) =>
+      value.fname.startsWith(hierarchy) &&
+      value.stub !== true &&
+      VaultUtils.isEqualV2(value.vault, vault)
+  );
+};
+
+/**
+ * This method check --vault argument. For a single vault workspace, if --vault not provided,
+ * returns the vault from workspace.
+ * For multi-vault workspace, if no --vault is given, returns an error, else returns selected vault
+ */
+const checkVaultArgs = (opts: {
+  engine: DEngineClient;
+  vaultName?: string;
+}) => {
+  const { engine, vaultName } = opts;
+  const { vaults } = engine;
+  if (_.size(vaults) > 1 && !vaultName) {
+    throw new DendronError({
+      message: "Please provide vault name in --vault arg",
+    });
+  } else {
+    return vaultName
+      ? VaultUtils.getVaultByNameOrThrow({ vaults: vaults, vname: vaultName })
```

Comment:
nitpick: eslint is warning me here for the shorthand `{vaults, vname: vaultName}`